### DISCUSSION TITLE
How can I specify different commit messages based on the dependency updated?

### DISCUSSION THREAD

**Discussion by andyinno** [renovatebot/renovate#20073] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.115

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitlab CE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I never saw this working... because I do not know if I can do this.

Basically I configured renovate with: ":semanticCommitTypeAll(fix)",

now I am in a situation where I have some regexManager that operate on custom C/C++ repos and other regexManagers that operates on gitlab-ci files.
I do not want that gitlab-ci files bumps the version number, I want that C/C++ custom regexManager bumps the version number. 

How can I specify to use fix() commit messages for C/C++ deps and chore() for gitlab-ci files?

At the moment my regexes are similar to those:

```
{
      "fileMatch": ["(^|\\/)package-lock\\.cmake$"],
      "matchStrings": [
          "cpmdeclarepackage\\s*\\(\\s*\\w+\\s*NAME\\s*(?<name>.*)\\s*GIT_TAG\\s*(?<currentValue>.*)\\s*GIT_REPOSITORY\\s*(?<depName>[^\\s)]+)\\s*\\)"
      ],
      "datasourceTemplate": "git-tags"      
    },
    {
      "fileMatch": ["^Dockerfile$"],
      "matchStrings": [
       "ARG\\sIMAGE=\"mygitlab-instance:443\\/mgroup\\/dependency_proxy\\/containers\\/(?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)\"\\s"
      ],
      "datasourceTemplate": "docker"
    }

```
Thank you in advance

### What you tried so far.

I tried to add 

```
":pathSemanticCommitType(*gitlab-ci.yml,chore)",
```

after

```
":semanticCommitTypeAll(fix)",
```

thinking that the second one overwrite the first, but this is not working

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by secustor** [comment, marked as answer]
You have to create a package rule which matches where you want to have a specific semantic commit type and then set it like we do in our own `renovate.json` https://github.com/renovatebot/renovate/blob/9d538a5210432024de2e329016bdd11175166fcd/renovate.json#L16-L19

**Reply by andyinno** [reply]
thank you for your reply.

I quite understand your reply but I absolutely ignore how to apply it.

How can I connect the regexp managers with the matchDepTypes in packageRule? 

I cannot understand from the doc how to achieve it

**Reply by secustor** [reply]
You are looking for [matchPackageManager](https://docs.renovatebot.com/configuration-options/#matchmanagers).
```
"matchPackageManager": ["regex"]
```

**Reply by andyinno** [reply]
ok, thanks

and what if I have multiple regex in my config?

like...

first regex => fix
second regex => chore?

**Reply by andyinno** [reply]
mmm... I just did this and it seems to work...

I do know know if it is something that is supported:

```
{
      "fileMatch": ["(^|\\/)package-lock\\.cmake$"],
      "matchStrings": [
          "cpmdeclarepackage\\s*\\(\\s*\\w+\\s*NAME\\s*(?<name>.*)\\s*GIT_TAG\\s*(?<currentValue>.*)\\s*GIT_REPOSITORY\\s*(?<depName>[^\\s)]+)\\s*\\)"
      ],
      "datasourceTemplate": "git-tags"  ,   
     "depTypeTemplate": "fix"
    },
    {
      "fileMatch": ["^Dockerfile$"],
      "matchStrings": [
       "ARG\\sIMAGE=\"mygitlab-instance:443\\/mgroup\\/dependency_proxy\\/containers\\/(?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)\"\\s"
      ],
      "datasourceTemplate": "docker",
      "depTypeTemplate": "chore"
    }
```

I run the bot and I found two updates, one with fix and one with chore.

**Reply by secustor** [reply]
> and what if I have multiple regex in my config?

In that case you have additional matchers. 

> I run the bot and I found two updates, one with fix and one with chore.

If this is not desired behaviour share a minimal reproduction repo.

**Reply by andyinno** [reply]
> > and what if I have multiple regex in my config?
> 
> In that case you have additional matchers.
Correct, but all regexes matches the same type. 

> 
> > I run the bot and I found two updates, one with fix and one with chore.
> 
> If this is not desired behaviour share a minimal reproduction repo.

My example is related to the regex definition. It works and it does what I want. I just don't know if depTypeTemplate can be any word or there is a list of what I can write there.

I have found the depType defined only for other managers. RegEx Manager does not define any depType. It says only that it is an optional argument. I tried to add a depTYpe named fix and chore in the two regex managers that I defined and the merge requests are created of fix and chore type.

My:

> I do know know if it is something that is supported

is related to the fix / chore  can and will always be supported as depType?

**Reply by secustor** [reply]
I referred with "additional matchers" to other `match*` functions https://docs.renovatebot.com/configuration-options/#matchpaths

> > I do know know if it is something that is supported
>
> is related to the fix / chore can and will always be supported as depType?

This is a simple string. As long the option to set them is available you can use this.

**Reply by andyinno** [reply]
Thank you for the confirmation!

have a nice week!

**Reply by viceice** [reply]
`depTypeTemplate` is valid inside regex manager
you can then use `matchDepType` in a package rule to assign the right semantic commit type.

https://docs.renovatebot.com/configuration-options/#deptypetemplate
